app/models/consistency.php

```diff
@@ -0,0 +1,79 @@
+    Project::getRepositoryLocales($repo),
+    $locale
+);
+
+/*
+    Since this is a very heavy view for Aurora (default repo), we display
+    an introductory text when reaching the page and use a paramter to
+    determine if we actually want to display the strings.
+*/
+$display_results = isset($_GET['display']) ? true : false;
+$reference_locale = Project::getReferenceLocale($repo);
+
+if ($display_results) {
+    // Get non-empty strings for reference and locale in this repository
+    $source_strings = array_filter(Utils::getRepoStrings($reference_locale, $repo), 'strlen');
+    $target_strings = array_filter(Utils::getRepoStrings($locale, $repo), 'strlen');
+
+    if (! empty($source_strings)) {
+        $duplicated_strings_english = Consistency::findDuplicates($source_strings);
+        $duplicated_strings_translation = Consistency::findDuplicates($target_strings);
+
+        // Filter English duplicates before comparing them with the localization
+        $duplicated_strings_english = Consistency::filterStrings($duplicated_strings_english, $repo);
+
+        // Get the strings duplicated in English but not in the localization
+        $missing_duplicates = array_diff_key($duplicated_strings_english, $duplicated_strings_translation);
+    }
+
+    if (! empty($duplicated_strings_english)) {
+        $inconsistent_translations = [];
+        foreach ($duplicated_strings_english as $key => $value) {
+            if (in_array($value, $missing_duplicates)) {
+                // Store inconsistency only if there's a translation available
+                    if (isset($target_strings[$key])) {
+                        $inconsistent_translations[] = [
+                            'source' => $source_strings[$key],
+                            'target' => $target_strings[$key],
+                        ];
+                    }
+            }
+        }
+
+        $strings_number = count($inconsistent_translations);
+        $inconsistencies = [];
+        if ($strings_number > 0) {
+            for ($i = 0; $i < $strings_number; $i++) {
+                $current_source = $inconsistent_translations[$i]['source'];
+                $available_translations = [$inconsistent_translations[$i]['target']];
+
+                // Get all translations for this English string
+                while ($i < $strings_number - 1 && $inconsistent_translations[$i + 1]['source'] == $current_source) {
+                    $i++;
+                    $available_translations[] = $inconsistent_translations[$i]['target'];
+                }
+
+                // Consider only unique different translations
+                $available_translations = array_unique($available_translations);
+
+                $inconsistencies[] = [
+                    'source' => $current_source,
+                    'target' => $available_translations,
+                ];
+            }
+        }
+    } else {
+        $strings_number = 0;
```

Comment:
The else clause adds complexity, especially since there are many ifs before and I had to look closely to see which if it was linked to. I think that since this is the default value, it should be set up early like line 23 with a comment indicating how it will be used in the view.